subsys/bluetooth/host/iso.c

```diff
@@ -442,15 +444,16 @@ static void bt_iso_chan_disconnected(struct bt_iso_chan *chan, uint8_t reason)
 #if defined(CONFIG_BT_ISO_UNICAST)
 			bool is_chan_connected;
 			struct bt_iso_cig *cig;
+			struct bt_iso_chan *tmp;
 
 			/* Update CIG state */
 			cig = get_cig(chan);
 			__ASSERT(cig != NULL, "CIG was NULL");
 
 			is_chan_connected = false;
-			SYS_SLIST_FOR_EACH_CONTAINER(&cig->cis_channels, chan, node) {
-				if (chan->state == BT_ISO_CONNECTED ||
-				chan->state == BT_ISO_CONNECT) {
+			SYS_SLIST_FOR_EACH_CONTAINER(&cig->cis_channels, tmp, node) {
```

Comment:
`SYS_SLIST_FOR_EACH_CONTAINER` will assign each item in the list to `tmp`. If you are familiar with something like python, this would be similar to `for tmp in cig->cis_channels`